commands/command_edit.cpp

```diff
@@ -1285,6 +1286,56 @@ void EditCylinder::undo()
 	redo();
 }
 
+EditSensors::EditSensors(int cylIndex, divecomputer *dcIn, std::vector<int> sensors)
+	: EditDivesBase(true), dc(dcIn)
+{
+	if (dives.empty())
+		return;
+
+	if (dives.size() == 1)
+		setText(Command::Base::tr("Edit sensors"));
+	else
+		throw std::runtime_error("Can only edit sensors for current dive computer");
+
+	for (int s = 0; s < MAX_SENSORS; ++s) {
+		if (std::find(sensors.begin(), sensors.end(), s) != sensors.end()) {
+			undoMapping.emplace_back(s, dc->sample[0].sensor[s]);
+			redoMapping.emplace_back(s, cylIndex);
+		} else if (dc->sample[0].sensor[s] == cylIndex) {
+			undoMapping.emplace_back(s, cylIndex);
+			redoMapping.emplace_back(s, NO_SENSOR);
+		}
```

Comment:
that code desperately needs some commenting. I've been staring at this for a couple of minutes and really don't know what you are doing here...

Reply:
Maybe the new comments clears things up a bit.

Reply:
I never understood the sensors code. I will just trust that this does the right thing.